packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3873,3 +3993,52 @@ class _CopySelectionAction extends ContextAction<CopySelectionTextIntent> {
   @override
   bool get isActionEnabled => state._value.selection.isValid && !state._value.selection.isCollapsed;
 }
+
+/// A class to return data for the `onContentCommitted` callback on text fields,
+/// which is when image content is inserted into the text field. The class holds
+/// information for the mime type, URI (location), and bytedata for the inserted
+/// content.
+@immutable
+class CommittedContent {
+  /// Creates a [CommittedContent] class. Any parameters can be null.
+  const CommittedContent({this.mimeType, this.uri, this.data});
+
+  /// mime type of inserted content
+  final String? mimeType;
+
+  /// URI (location) of inserted content
+  final String? uri;
+
+  /// Bytedata of inserted content
+  final Uint8List? data;
+
+  /// Convenience getter to check if bytedata is available for the inserted content
+  bool get hasData => data != null && data!.isNotEmpty;
+
+  /// Converts Map received from Flutter Engine into the Dart class
+  static CommittedContent fromMap(Map<String, dynamic>? data) {
```

Comment:
Could this be a named constructor?

Reply:
There is already a constructor for the class.. Sid you mean just make `data` a named parameter for the fromMap method? Or did you mean replace the existing constructor?